File: src/main/java/me/leodev/trollingfreedomreborn/trolls/inventory/LockInventory.java
package me.leodev.trollingfreedomreborn.trolls.inventory;

import me.leodev.trollingfreedomreborn.main.Core;
import org.bukkit.entity.Player;
import org.bukkit.event.Listener;
import org.bukkit.scheduler.BukkitRunnable;

public class LockInventory implements Listener {


    public static void Lock(Player p) {
        // 1. Define the BukkitRunnable separately
        BukkitRunnable runnable = new BukkitRunnable() {
            @Override
            public void run() {
                if (!p.isOnline()) {
                    this.cancel();
                    return;
                }

                p.closeInventory();

                // Note: If you want UnTroll to handle the stopping,
                // you usually don't need the sneaking check here.
                if (p.isSneaking()) {
                    this.cancel();
                }
            }
        };

        // 2. Start the task using the Runnable's own method
        // This returns a BukkitTask object, from which we get the ID
        int taskId = runnable.runTaskTimer(Core.instance, 1L, 1L).getTaskId();

        // 3. Register the task for your surgical UnTroll system
        Core.instance.addTask(p, "lockinventory", taskId);
    }
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/trolls/inventory/LockInventory.kt
package com.leomadrassi.trollingfreedomreborn.trolls.inventory

import com.leomadrassi.trollingfreedomreborn.main.Core
import org.bukkit.entity.Player
import org.bukkit.event.Listener
import org.bukkit.scheduler.BukkitRunnable

class LockInventory : Listener {
    companion object {
        @JvmStatic
        fun Lock(p: Player) {
            val runnable = object : BukkitRunnable() {
                override fun run() {
                    if (!p.isOnline) {
                        cancel()
                        return
                    }

                    p.closeInventory()

                    if (p.isSneaking) {
                        cancel()
                    }
                }
            }

            val taskId = runnable.runTaskTimer(Core.instance, 1L, 1L).taskId
            Core.instance.addTask(p, "lockinventory", taskId)
        }
    }
}
